src/web/app/components/view-results-panel/view-results-panel.component.ts

```diff
@@ -0,0 +1,158 @@
+  ResponseVisibleSetting,
+  SessionVisibleSetting,
+} from '../../../types/api-output';
+
+/**
+ * Displaying the view results panel.
+ */
+@Component({
+  selector: 'tm-view-results-panel',
+  templateUrl: './view-results-panel.component.html',
+  styleUrls: ['./view-results-panel.component.scss'],
+})
+export class ViewResultsPanelComponent implements OnInit {
+
+  // enum
+  InstructorSessionResultSectionType: typeof InstructorSessionResultSectionType = InstructorSessionResultSectionType;
+  InstructorSessionResultViewType: typeof InstructorSessionResultViewType = InstructorSessionResultViewType;
+
+  @Input()
+  session: FeedbackSession = {
+    courseId: '',
+    timeZone: '',
+    feedbackSessionName: '',
+    instructions: '',
+    submissionStartTimestamp: 0,
+    submissionEndTimestamp: 0,
+    gracePeriod: 0,
+    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+    publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
+    isClosingEmailEnabled: true,
+    isPublishedEmailEnabled: true,
+    createdAtTimestamp: 0,
+  };
+
+  @Input()
+  viewTooltipText: string = 'View results in different formats';
+
+  @Input()
+  viewType: string = InstructorSessionResultViewType.QUESTION;
+
+  @Input()
+  section: string = '';
+
+  @Input()
+  sectionsModel: Record<string, SectionTabModel> = {};
+
+  @Input()
+  sectionType: InstructorSessionResultSectionType = InstructorSessionResultSectionType.EITHER;
+
+  @Input()
+  groupByTeam: boolean = true;
+
+  @Input()
+  showStatistics: boolean = true;
+
+  @Input()
+  indicateMissingResponses: boolean = true;
+
+  @Output()
+  sectionChange: EventEmitter<string> = new EventEmitter<string>();
+
+  @Output()
+  viewTypeChange: EventEmitter<string> = new EventEmitter<string>();
+
+  @Output()
+  viewTooltipTextChange: EventEmitter<string> = new EventEmitter<string>();
+
+  @Output()
+  sectionTypeChange: EventEmitter<InstructorSessionResultSectionType> =
+    new EventEmitter<InstructorSessionResultSectionType>();
+
+  @Output()
+  groupByTeamChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  @Output()
+  showStatisticsChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  @Output()
+  indicateMissingResponsesChange: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  @Output()
+  collapseAllTabsEvent: EventEmitter<void> = new EventEmitter<void>();
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  collapseAllTabsHandler(): void {
+    this.collapseAllTabsEvent.emit();
+  }
+
+  /**
+   * Handles view type changes.
+   */
+  handleViewTypeChange(newViewType: InstructorSessionResultViewType): void {
+    if (this.viewType === newViewType) {
+      // do nothing
+      return;
+    }
+    this.viewTypeChange.emit(newViewType);
+
+    // change tooltip text based on currently selected view type
+    switch (this.viewType) {
```

Comment:
There are 5 view types but only 3 is covered here

Reply:
fixed in f4952ca